### ISSUE TITLE
--prod doesn't work on Angular CLI projects

### ISSUE THREAD

**Issue by chadbr** [OSSIndex/auditjs#25]
I'm trying to use auditjs with the --prod flag on an angular cli project, and it fails to report anything.

Steps to repo:

```
npm install -g @angular/cli
ng new myapp
cd myapp
npm install auditjs --save-dev
```

running `auditjs --prod` yields nothing...

Thanks, Chad

https://github.com/angular/angular-cli/issues/8644#issuecomment-348215788

**Comment by OSSIndex-Admin** [comment]
Thanks for the report and the steps to reproduce the problem. I will attempt to resolve this issue shortly.

**Comment by ken-duck** [comment]
Very curious. I have done the same, and receive different results.

```
bragi:myapp twoducks$ pwd
.../issue-25/myapp

bragi:myapp twoducks$ ls
README.md		e2e			node_modules		package.json		src
angular-cli.json	karma.conf.js		package-lock.json	protractor.conf.js	tslint.json

bragi:myapp twoducks$ auditjs --prod
[1/14] nodejs v8.2.1   30 known vulnerabilities, 0 affecting installed version
[2/14] @angular/common 2.4.10   No known vulnerabilities...
[3/14] @angular/compiler 2.4.10   No known vulnerabilities...
[4/14] @angular/core 2.4.10   No known vulnerabilities...
[5/14] @angular/forms 2.4.10   No known vulnerabilities...
[6/14] @angular/http 2.4.10   No known vulnerabilities...
[7/14] @angular/platform-browser 2.4.10   No known vulnerabilities...
[8/14] @angular/platform-browser-dynamic 2.4.10   No known vulnerabilities...
[9/14] @angular/router 3.4.10   No known vulnerabilities...
[10/14] core-js 2.5.1   No known vulnerabilities...
[11/14] rxjs 5.5.3   No known vulnerabilities...
[12/14] symbol-observable 1.1.0   No known vulnerabilities...
[13/14] ts-helpers 1.1.2   No known vulnerabilities...
[14/14] zone.js 0.7.8   No known vulnerabilities...

Audited dependencies: 14, Vulnerabilities: 0, Ignored: 0
```

Could you perhaps send some information:
* an `ls` for your myapp directory
* an `ls` of the node_modules directory
* your auditjs version (I did a recent upgrade, but it should not make too big a difference)

**Comment by chadbr** [comment]
@ken-duck it looks like you're using an older version of `@angular/cli` -- you should be seeing version 5 angular components.

I uploaded the repo here: https://github.com/chadbr/myapp

```
Chads-MBP:tmp chadbr$ npm install -g @angular/cli
ng new myapp
cd myapp
/usr/local/bin/ng -> /usr/local/lib/node_modules/@angular/cli/bin/ng
+ @angular/cli@1.5.5
updated 6 packages in 22.916s
Chads-MBP:tmp chadbr$ ng new myapp
cd myapp
npm install auditjs --save-devUnable to find "@angular/cli" in devDependencies.

Please take the following steps to avoid issues:
"npm install --save-dev @angular/cli@latest" 

  create myapp/README.md (1021 bytes)
  create myapp/.angular-cli.json (1240 bytes)
  create myapp/.editorconfig (245 bytes)
  create myapp/.gitignore (516 bytes)
  create myapp/src/assets/.gitkeep (0 bytes)
  create myapp/src/environments/environment.prod.ts (51 bytes)
  create myapp/src/environments/environment.ts (387 bytes)
  create myapp/src/favicon.ico (5430 bytes)
  create myapp/src/index.html (292 bytes)
  create myapp/src/main.ts (370 bytes)
  create myapp/src/polyfills.ts (2405 bytes)
  create myapp/src/styles.css (80 bytes)
  create myapp/src/test.ts (1085 bytes)
  create myapp/src/tsconfig.app.json (211 bytes)
  create myapp/src/tsconfig.spec.json (304 bytes)
  create myapp/src/typings.d.ts (104 bytes)
  create myapp/e2e/app.e2e-spec.ts (287 bytes)
  create myapp/e2e/app.po.ts (208 bytes)
  create myapp/e2e/tsconfig.e2e.json (235 bytes)
  create myapp/karma.conf.js (923 bytes)
  create myapp/package.json (1310 bytes)
  create myapp/protractor.conf.js (722 bytes)
  create myapp/tsconfig.json (363 bytes)
  create myapp/tslint.json (3097 bytes)
  create myapp/src/app/app.module.ts (316 bytes)
  create myapp/src/app/app.component.css (0 bytes)
  create myapp/src/app/app.component.html (1141 bytes)
  create myapp/src/app/app.component.spec.ts (986 bytes)
  create myapp/src/app/app.component.ts (207 bytes)
Installing packages for tooling via yarn.
Installed packages for tooling via yarn.
Successfully initialized git.
Project 'myapp' successfully created.
Chads-MBP:tmp chadbr$ cd myapp
Chads-MBP:myapp chadbr$ npm install auditjs --save-dev
npm WARN gentlyRm not removing /Users/chadbr/dev/vsts/tmp/myapp/node_modules/.bin/semver as it wasn't installed by /Users/chadbr/dev/vsts/tmp/myapp/node_modules/semver
npm WARN gentlyRm not removing /Users/chadbr/dev/vsts/tmp/myapp/node_modules/.bin/webdriver-manager as it wasn't installed by /Users/chadbr/dev/vsts/tmp/myapp/node_modules/protractor
npm WARN gentlyRm not removing /Users/chadbr/dev/vsts/tmp/myapp/node_modules/.bin/uglifyjs as it wasn't installed by /Users/chadbr/dev/vsts/tmp/myapp/node_modules/uglify-js
npm WARN gentlyRm not removing /Users/chadbr/dev/vsts/tmp/myapp/node_modules/.bin/nopt as it wasn't installed by /Users/chadbr/dev/vsts/tmp/myapp/node_modules/nopt
npm WARN gentlyRm not removing /Users/chadbr/dev/vsts/tmp/myapp/node_modules/fsevents/node_modules/.bin/node-pre-gyp as it wasn't installed by /Users/chadbr/dev/vsts/tmp/myapp/node_modules/fsevents/node_modules/node-pre-gyp
npm notice created a lockfile as package-lock.json. You should commit this file.
+ auditjs@2.2.8
added 312 packages, removed 1143 packages, updated 13 packages and moved 1 package in 17.337s
Chads-MBP:myapp chadbr$ auditjs --prod
[1/1] nodejs v8.7.0   30 known vulnerabilities, 0 affecting installed version

Audited dependencies: 1, Vulnerabilities: 0, Ignored: 0

Chads-MBP:myapp chadbr$ 
```

**Comment by OSSIndex-Admin** [comment]
Short update: I have installed your test and can see the same issue. I can definitely see some likely problems.

**Comment by OSSIndex-Admin** [comment]
You managed to trigger a couple of bugs with that test. I have resolved the issue (I hope) and published 2.2.9 on npm.
